Sources/Commands/SwiftTestTool.swift

```diff
@@ -280,20 +288,51 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
                 numJobs: options.numberOfWorkers ?? ProcessInfo.processInfo.activeProcessorCount,
                 diagnostics: diagnostics,
                 options: self.options,
-                buildParameters: try self.buildParameters()
+                buildParameters: buildParameters
             )
             try runner.run(tests)
 
             if !runner.ranSuccessfully {
                 executionStatus = .failure
             }
         }
+
+        if options.shouldEnableCodeCoverage {
+            let buildParameters = try self.buildParameters()
+            let toolChain = try getToolchain()
+            try mergeRawDataFiles(fs: fileManager, path: buildParameters.buildPath.appending(component: "codecov"), toolChain: toolChain)
+            let graph = try loadPackageGraph()
+            for packages in graph.rootPackages {
+                let testPath = try buildTestsIfNeeded(options, graph: graph, binary: true)
+                try generateReport(fs: fileManager, testPath: testPath, buildPath: buildParameters.buildPath, name: packages.name, toolChain: toolChain)
+            }
+        }
+    }
+
+    private func mergeRawDataFiles(fs: FileSystem, path: AbsolutePath, toolChain: UserToolchain) throws {
+            let files = try fs.getDirectoryContents(path)
+        let llvmProf = try toolChain.getTool(toolName: "llvm-profdata")
+            var args = [llvmProf.asString, "merge", "-sparse"]
+            for file in files {
+                args.append(path.appending(component: file).asString)
+            }
+            args += ["-o",  path.appending(component: "default.profdata").asString]
+            try Process.checkNonZeroExit(arguments: args)
+    }
+
+    private func generateReport(fs: FileSystem, testPath: AbsolutePath, buildPath: AbsolutePath, name: String, toolChain: UserToolchain) throws {
```

Comment:
Same comment as above, we don't need to pass most of these arguments. The method name also seems outdated given this method generates json now.

Reply:
done